src/thor/trippathbuilder.cc

```diff
@@ -1041,16 +1039,21 @@ TripPathBuilder::Build(const AttributesController& controller,
       float edge_length = static_cast<float>(directededge->length());
       TrimShape(edge_shape, edge_begin_info.distance_along * edge_length, edge_begin_info.vertex,
                 edge_end_info.distance_along * edge_length, edge_end_info.vertex);
-
       // Add edge shape to trip
       trip_shape.insert(trip_shape.end(),
                         (edge_shape.begin() + ((edge_begin_info.exists || is_first_edge) ? 0 : 1)),
                         edge_shape.end());
 
-      // If edge_begin_info.exists and not the first edge
-      // then increment begin_index
-      // since the previous end shape index should not equal
-      // the current begin shape index because of discontinuity
+      // Adjust the length of the trimmed edge if this is a route (check if through_location exist).
+      // Do not want to trim the edge length if this is part of map matching (TODO - verify).
+      if (through_loc.size() > 0) {
+        trip_edge->set_length(edge_length * kKmPerMeter *
+                              (edge_end_info.distance_along - edge_begin_info.distance_along));
+      }
```

Comment:
here's the bit in TPB that says oh, its not map match, which means we should set the length of the edge to the actual portion the route uses (in map matching we send the length of the full edge since its a meta-data look up service)